src/Nethermind/Nethermind.Network/Rlpx/ZeroFrameDecoder.cs

```diff
@@ -34,6 +34,12 @@ public ZeroFrameDecoder(IFrameCipher frameCipher, IFrameMacProcessor frameMacPro
             _logger = logManager?.GetClassLogger<ZeroFrameDecoder>() ?? throw new ArgumentNullException(nameof(logManager));
         }
 
+        public override void HandlerRemoved(IChannelHandlerContext context)
+        {
+            base.HandlerRemoved(context);
+            _innerBuffer?.Release();
```

Comment:
Are you sure it isn't released anywhere else?

Reply:
Its a class field that is not sent to netty yet, otherwise it would be null. So its not releasing it automatically.

Reply:
it is added to `output` which is later used by dotnetty code and it sometimes releases stuff, so hence my question